commandhandler.go

```diff
@@ -0,0 +1,556 @@
+package main
+
+import (
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/acarl005/stripansi"
+	"github.com/shurcooL/tictactoe"
+)
+
+type commandInfo struct {
+	name          string
+	description   string
+	callable      func(user *user, args []string)
+	echo          bool
+	requiresAdmin bool
+	aliases       []string
+}
+
+var (
+	commands = make([]commandInfo, 0)
+)
+
+func processMessage(u *user, message string) {
+	message = strings.TrimSpace(message)
+
+	if message == "" {
+		return // Don't allow empty messages
+	}
+	splitted := strings.Split(message, " ")
+	if strings.HasPrefix(splitted[0], "./") {
+		if u.slack {
+			u.room.broadcast(devbot, "Slack users can't use commands", false)
```

Comment:
we want slack users to be able to use commands, maybe add a new field in `commandInfo` saying whether slack is allowed to use it or not.

Reply:
I mean this is to avoid the spam in chat we had earlier

Reply:
what spam?

Reply:
people spamming ./color random etc

Reply:
this prevents slack users from using commands, how will it prevent people from spamming ./color random lol

Reply:
well not ./color exactly but things like commands help etc giant blobs of text. Theres no filtering on what slack users say and we have to respond with in public chat about it.

Reply:
@TAG-Epic we need this

Reply:
Also add a bool in comamndInfo saying whether slack users can run a certain command